### PR TITLE
Add bfloat16 support to upsample

### LINKED ISSUE #80339: BFloat16 support for upsampling on CUDA
### 🚀 The feature, motivation and pitch

Upsampling operations currently do not support bfloat16 on CUDA, as discussed in https://github.com/pytorch/pytorch/issues/57707

### Alternatives

_No response_

### Additional context

_No response_

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are @ngimel

### PR DESCRIPTION
**Pull request by jamaliki**
Fixes https://github.com/pytorch/pytorch/issues/80339

This PR was previously here: https://github.com/pytorch/pytorch/pull/95159

### CODE REVIEW

**Comment by jamaliki** [comment]
@eqy I think this is good to go now

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -5308,7 +5308,6 @@ def sample_inputs_std_var(op_info, device, dtype, requires_grad, **kwargs):
     yield SampleInput(tensor_1d(), dim=0, unbiased=True, keepdim=True)
     yield SampleInput(tensor_1d(), dim=0, unbiased=False, keepdim=False)
 
-    yield SampleInput(tensor_nd(), dim=(1,), correction=1.3)
```
Comment (ezyang) [inline torch/testing/_internal/common_methods_invocations.py:5311]: how come?

**Review by ezyang — APPROVED** [review]

**Comment by mreyarea-fy** [comment]
I just built the nightly version but this is not available yet. When will it be added?

**Comment by jamaliki** [comment]
@mreyarea-fy I just installed `'2.1.0.dev20230307+cu117'` and it ran just fine

**Comment by Cyberbeing** [comment]
It would appear this change did not make it into the final Torch 2.0.0 build released the other day, which is unfortunate. 

I was hopeful, with the first nightly it appeared in being torch-2.0.0.dev20230228, but I guess it must have been committed too close to 2.0.0 release cut-off to be included.

Does this mean it now won't appear in a release build until Torch 2.1.0, or is there some other problem which is currently holding it back from being included in a release build?

**Comment by jamaliki** [comment]
Yes I also don't think it made it to Pytorch-2.0. I know it is in Pytorch-2.1

**Pull request closed without merging**